feat: Switch Slack Channel links to the CNCF Slack (#7304)
cloudnative-pg/cloudnative-pg: docs/src/supported_releases.md

We offer two types of support:

Technical support
:   Technical assistance is provided on a best-effort basis and only for supported releases.
    You can request help from the community through the
    [#cloudnativepg-users](https://cloud-native.slack.com/archives/C08MAUJ7NPM) channel
    on the CNCF Slack workspace (if you're not yet a member, you can join via the
    [Community Inviter App](https://communityinviter.com/apps/cloud-native/cncf))

>     or through [GitHub Discussions](https://github.com/cloudnative-pg/cloudnative-pg/discussions)

Reviewer: Here too I'd refer to the Communications section of the README.

Reviewer: ```suggestion
    You can request help from the community through the CNCF Slack worspace
    or GitHub Discussions.
    We refer you to our [Communication channels](../../README.md#communications)
```

Reviewer 2: I had a similar idea Jaime, but it was primarily influenced by the fact that URLs would regularly change in the past. Now, with the CNCF Slack, these links will become immutable, so I believe it is fine.
